app/cafe.py

```diff
@@ -0,0 +1,25 @@
+from datetime import date
+from app.errors import (
+    NotVaccinatedError, OutdatedVaccineError, NotWearingMaskError
+)
+
+
+class Cafe:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def visit_cafe(self, visitor: dict) -> str:
+        if not visitor.get("vaccine"):
+            raise NotVaccinatedError(
+                "You need to be vaccinated to visit the cafe!"
+            )
+        expiration_date = visitor["vaccine"].get("expiration_date")
+        if expiration_date and expiration_date < date.today():

```

Comment:
Code Efficiency: You are checking if `expiration_date` is not None and then comparing it with `date.today()`. You can simplify this by using a single line `if not expiration_date or expiration_date < date.today():`. This way, if `expiration_date` is None, Python's short-circuiting behavior will prevent the second condition from being evaluated.